mongospecs/base.py

```
@@ -1,852 +1,26 @@
-from contextlib import contextmanager
 from copy import deepcopy
 
-from blinker import signal
-from bson import BSON, ObjectId
-from pymongo import ASCENDING, MongoClient, UpdateOne
-from pymongo.client_session import ClientSession
-from pymongo.collection import Collection
-from pymongo.database import Database
-from pymongo.errors import ConnectionFailure, OperationFailure
-from typing_extensions import Self
+from mongospecs.types import SpecBaseType, SubSpecBaseType
```

Ruff (0.8.2)

`mongospecs.types.SpecBaseType` imported but unused

Remove unused import: `mongospecs.types.SpecBaseType`

(F401)